packages/medusa/src/api-v2/store/carts/route.ts

```diff
@@ -1,23 +1,31 @@
-import { createCartsWorkflow } from "@medusajs/core-flows"
+import { createCartWorkflow } from "@medusajs/core-flows"
 import { CreateCartDTO } from "@medusajs/types"
 import { MedusaRequest, MedusaResponse } from "../../../types/routing"
+import { defaultStoreCartRemoteQueryObject } from "../carts/query-config"
 
 export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
-  const createCartWorkflow = createCartsWorkflow(req.scope)
-  const cartData = [
-    {
-      ...(req.validatedBody as CreateCartDTO),
-    },
-  ]
+  const workflow = createCartWorkflow(req.scope)
 
-  const { result, errors } = await createCartWorkflow.run({
-    input: { cartData },
+  const { result, errors } = await workflow.run({
+    input: { cartData: req.validatedBody as CreateCartDTO },
     throwOnError: false,
   })
 
   if (Array.isArray(errors) && errors[0]) {
     throw errors[0].error
   }
 
-  res.status(200).json({ cart: result[0] })
+  const remoteQuery = req.scope.resolve("remoteQuery")
+
+  const variables = { id: result[0].id }
```

Comment:
Is it worth asserting the result length in such cases?

Reply:
theoratically at this point, there will always be an item in the collection